internal/renderers/skia/lib.rs

```diff
@@ -365,36 +457,59 @@ impl SkiaRenderer {
                 }
             }
 
+            if let Some(callback) = self.rendering_notifier.borrow_mut().as_mut() {
+                // For the BeforeRendering rendering notifier callback it's important that this happens *after* clearing
+                // the back buffer, in order to allow the callback to provide its own rendering of the background.
+                // Skia's clear() will merely schedule a clear call, so flush right away to make it immediate.
+                if let Some(ctx) = gr_context.as_mut() {
+                    ctx.flush(None);
+                }
+
+                if let Some(surface) = surface {
+                    surface.with_graphics_api(&mut |api| {
+                        callback.notify(RenderingState::BeforeRendering, &api)
+                    })
+                }
+            }
+
             for (component, origin) in components {
                 i_slint_core::item_rendering::render_component_items(
                     component,
-                    &mut item_renderer,
+                    item_renderer,
                     *origin,
                 );
             }
 
-            if let Some(collector) = &self.rendering_metrics_collector.borrow_mut().as_ref() {
-                collector.measure_frame_rendered(&mut item_renderer);
+            if let Some(path) = dirty_region_to_visualize {
```

Comment:
In the software renderer i left the code commented out:
https://github.com/slint-ui/slint/blob/9d851b7942efe89bd59eaa15a5f033b2015bedf2/internal/core/software_renderer.rs#L1057
But your option with the env variable make sense.